#! /bin/sh -e

# All lines beginning with `# DPATCH:' are a description of the patch.
# DP: This patch fixes a bug into ostream::operator<<(double) due to the wrong size
# DP: passed into the __convert_from_v method. The wrong size is then passed to
# DP: std::snprintf function, that, on uClibc, doens't handle sized 0 buffer.

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p1 < $0
        #cd ${dir}gcc && autoconf
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p1 < $0
        #rm ${dir}gcc/configure
        ;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1
esac
exit 0

Signed-off-by: Carmelo Amoroso <carmelo.amoroso@st.com>

Index: gcc-4.3.1/libstdc++-v3/include/bits/locale_facets.tcc
===================================================================
--- gcc-4.3.1.orig/libstdc++-v3/include/bits/locale_facets.tcc	2007-11-26 17:59:41.000000000 -0800
+++ gcc-4.3.1/libstdc++-v3/include/bits/locale_facets.tcc	2008-08-16 02:14:48.000000000 -0700
@@ -1004,7 +1004,7 @@
 	const int __cs_size = __fixed ? __max_exp + __prec + 4
 	                              : __max_digits * 2 + __prec;
 	char* __cs = static_cast<char*>(__builtin_alloca(__cs_size));
-	__len = std::__convert_from_v(_S_get_c_locale(), __cs, 0, __fbuf, 
+	__len = std::__convert_from_v(_S_get_c_locale(), __cs, __cs_size, __fbuf,
 				      __prec, __v);
 #endif
 
